cabra/src/pages/TaskEdit.tsx

```diff
@@ -0,0 +1,59 @@
+import "twin.macro";
+
+import TaskForm, { TaskFormInputs } from "./components/TaskForm";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { useNavigate, useParams } from "react-router";
+
+import { ArrowLeftIcon } from "@heroicons/react/solid";
+import { ITask } from "../types/task";
+import Layout from "./components/Layout";
+import NavigationButton from "./components/NavigationButton";
+import axios from "axios";
+import { routeHelpers } from "../routes";
+
+export default function TaskEditPage() {
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const client = useQueryClient();
+  const { data, isLoading } = useQuery(
+    ["task", id],
+    () => axios.get<ITask>(`/tasks/${id}`),
+    {
+      onError: () => navigate(routeHelpers.notFound, { replace: true }),
+      retry: 1,
+    }
+  );
+
+  const addTask = useMutation((task: TaskFormInputs) =>
+    axios.put<ITask>(`/tasks/${id}`, task)
+  );
+
+  const onSubmit = (variables: TaskFormInputs) =>
+    addTask.mutateAsync(variables, {
+      onSuccess: () => {
+        client.invalidateQueries("tasks");
+        client.invalidateQueries(["tasks", id]);
```

Comment:
Maybe we should extract React query keys to a constant? Wdyt?

Reply:
Yeah, and we should probably gather them with related HTTP calls